### ISSUE TITLE
Touch doesnt work after Lockscreen

### ISSUE THREAD

**Issue by SaulGoodman88** [AOSP-JF-MM/issue_tracking_aosp#12]
after lockscreen my touch doenst work anymore and i have to reboot the phone.
same issue with optimized cm 12.1 rom, but with other kernels the issue disappears.
iam using another cm13 build at the moment and touch works but i really like your roms.
if you need a log i can flash your rom again, the issue appears everytime.

```
Phone model: jfltexx GT-I9505
Baseband version: ex. I9505XXUHOH2
ROM version: ex. 20151206

GApps Package: Open GApps [Nano] [20151212]

Did you do a clean install for the currently installed build? yes
Did you reboot after having the issue? yes
Did you do any system modification? no
Did you flash another kernel? no
Did you install Xposed framework? no
```

**Comment by AntaresOne** [comment]
It's a known issue by the Alucard kernel as it's the only common thing across our 5.1 ROMs and AOSP 6.0. Yes, logs would be great rightly after you experience it, make sure you get (with ADB):
- adb logcat -v long > logcat.txt
- adb shell cat /proc/kmsg > /sdcard/kmsg.txt (stop this after 5 seconds by pressing CTRL+C)

**Comment by SaulGoodman88** [comment]
thank you for the answer.
i tried to do a logcat, not sure if i did it right.
i can attach the logcat.txt but adb shell cat /proc/kmsg > /sdcard/kmsg.txt doesnt work it says "The system can not find the path specified ."

hope this is what you need.

[logcat.txt](https://github.com/AOSP-JF-MM/issue_tracking_aosp/files/62996/logcat.txt)

**Comment by SaulGoodman88** [comment]
by the way i used TWRP v2.8.7.2 to flash the rom and did a clean flash.

**Comment by SaulGoodman88** [comment]
could this be the Problem? 

[ 12-15 18:49:35.392   814:  842 I/DisplayManagerService ]

Display device changed state: "Integrierter Bildschirm", OFF

**Comment by Zyphrex** [comment]
I too am having the same issue. I had the issue with 5.1 also and I believe I flashed a different kernel to the device which corrected the issue entirely. I just don't remember what kernel I actually used to fix the issue in 5.1.

It's odd that it doesn't happen when the device boots for the first time... Only when waking the device up will the touch screen become unresponsive. It's definitely a kernel issue as it happens in both system and recovery.

**Comment by SaulGoodman88** [comment]
yes it´s definitely a kernel issue. in optmized cm12.1 Rom i used the SaberMod Kernel and it fixed the issue, but i really want to use the alucard kernel because for me it´s the best kernel. battery life was awesome in the past. really hope alucard can fix this issue.

**Comment by SaulGoodman88** [comment]
@AntaresOne i have an idea what could be the problem that causes this issue. After my Touchscreen was broken i changed it but the new one is an touchscreen from the deep black version (should be the I9515) and i have the I9505 Device.

Could it be that both devices use different Touch Controllers? I have read something about that flashing an I9505 kernel on an I9515 Device causes the unresponsive Touchscreen issue. Just wondering why the Stock Cm Kernel works.

hope you understand my bad english

**Comment by smeroni68** [comment]
One of the differences from 9505 and 9515 is the touch... this is one of the hardware reasons why our unified roms DO NOT support 9515...

**Comment by SaulGoodman88** [comment]
Didn't know that, any chance to get the Alucard Kernel working with this touch on my Device? CyanogenMod Kernel works fine. Thank you for the Answer @smeroni68.

**Comment by alfredkrohmer** [comment]
@SaulGoodman88 I own the S4 Black Edition (if this is what you mean with "Deep Black Edition") and it's a I9505 (at least my Status menu says so and Google results say so too). I also had the touchscreen problem on the Lollipop ROM with the Alucard kernel, but now I'm using the the official kernel provided by with the latest Lollipop ROM in the xda thread and the problem is gone.

**Comment by SaulGoodman88** [comment]
@devkid yes this is the Device what i mean, i thought it was I9515. All Cm12.1 and Cm13 ROMs with Stock Kernel are working fine, i only have this issues with the Alucard Kernel. On my other Device everthing works, so it must be a problem with the Digitizer.

Would be nice if the Devs can fix this, because i really like the Alucard Kernel.

**Comment by AntaresOne** [comment]
@SaulGoodman88 what has been changed there? full display assembly or just the glass? The touch screen controller is on the assembly, therefore if you change it with an I9515 one neither with stock ROMs will work... Does that happens rightly after you boot or you can unlock the lockscreen at least 1 time?

**Comment by alfredkrohmer** [comment]
@AntaresOne Not speaking for SaulGoodman88, but in my case the touchscreen stopped working randomly when waking up the device to the lockscreen; not every time, sometimes it worked for days straight, but eventually it would occur.

What I noticed was that there was a "gap" in the battery stats history in the time that the touchscreen was unresponsive. (no data or 0%? there was just a gap in the graph) In the time before being unresponsive and after a reboot, the battery stats would be shown normally.

**Comment by SaulGoodman88** [comment]
@AntaresOne  i changed full display assembly. 

It happens when i lock the screen for the first time, when i start the phone i can enter the pin and use it, but when i lock the phone the touch is unresponsive and i must hard reset the phone. 

With the August Build from optimized 12.1 Rom sometimes i can unlock the phone and sometimes not. On October Builds and AOSP 6.0 Builds i never can unlock the phone.

**Comment by CidK** [comment]
I am having the exact same issue, i even tried the latest 5.1 build and am getting the same behavior. And for me this touch screen non responsive happens all the time and every time the phone goes to sleep. Hard Reboot is required for me to continue using the phone. 

Even the Cyanogen roms are giving me this issue, the only custom roms i was able to get working was from Tipsy (Tesla-jflte-5.1.1.Arc_Flash.2.3-MR1-20151013).

I have a S4 Canadian version SGH-I337m. 

Is there anything i can do to fix this ?

**Comment by alfredkrohmer** [comment]
@CidK Have you tried flashing the official [CM kernel](https://romhut.com/addons/cm-kernel_20151130) together with the 5.1 ROM? This fixed the problem for me.

**Comment by CidK** [comment]
I will try this right away and keep you posted.

**Comment by B--B** [comment]
Please try our latest release, many things has been fixed on kernel side, hopefully even this issue has been fixed...! Sorry but we cannot test by ourselves, because none of us has this kind of issue...!

**Comment by SaulGoodman88** [comment]
ok i will try the latest release and report if the issue is gone

**Comment by SaulGoodman88** [comment]
same problem :( with unofficial and official cm13 roms i don't have this issue.

**Comment by B--B** [comment]
I've uploaded the stock kernel as addons, please give it a try and let us know if helps...!

Download link: https://romhut.com/addons/stock_kernel-183fa1da-4086-4188-b97a-10e74b38cff2

Please not that flashing this kernel will break exFAT and NTFS support so, be careful if your sdcard is exFAT or NTFS!

**Comment by SaulGoodman88** [comment]
Thank you! With the stock kernel everything works fine now, no more freezes.
Would be nice if you can find out what's the problem for some devices with Alucard kernel and fix it some day. :)

**Comment by micjon** [comment]
I have the same problem. If I flash ROM (AOSP 6.0) without Gapps, then touch screen works.
But if I then flash Gapps, touch screen again stops working. Touch screen is also after replacement. I have I9505.

**Comment by Dionizasd** [comment]
I have the same problem, also screen was changed (i9505)
With stock kernel everything is fine. Regards

**Comment by CidK** [comment]
Now am having issues with not being able to hear anything while taking or
receiving a call.

It happens every 3 or 4 calls, i need to reboot the phone in order to place
a call, and then it stop working again after 3 or 4 calls.

Its been happening on every CM13 and CM12,1

**Comment by crispychiken** [comment]
I'm getting the same issue as SaulGoodman88, after the display goes off I turn it back on and touching the screen does nothing. I tested it with a lockscreen and without, getting the same results either way. 
I'm using:
Cricket Galaxy S4 SGH-I337Z
Optimized-LineageOS-14.1-20170312-Stable1
open_gapps-arm-7.1-pico-20170402
TWRP 2.8.4.0

TWRP also had this issue but I worked around it by not letting the screen turn off >_>

**Comment by AthensTxMc** [comment]
I found a way around this but created another .. If you flash the stock rom it will fix the unresponsive touch screen but it breaks the bluetooth .. that is on the jdc 7.1,2 on a m919 .. I have flashed everything fresh and unless I go to addons on the OTA updates and flash the stock kernel the phone will only respond to the volume, power left. and right screen buttons but the main swipe screen is untouchable! I even turned off all security and set the swipe to none hoping that would help, tried setting a PIN and a pattern but couldn't access any of those either .. If I flash the stock kernel I get a persistent "bluetooth sharing has stopped" or something error (sorry I have been doing this for 7 hours and my wife needs her phone to work) and I have to mute that in order to do anything at all .. Anyway, in general I am surprised it works as well as it does and I appreciate all the work that went into this ..
